### PR TITLE
Reconciler split

### PR DESCRIPTION
**Pull request by cppforlife**
- moved reconcilers to their related packages

(easier to review by commits.)

### CODE REVIEW

File: pkg/app/apps_reconciler_test.go [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-package controller_test
+package app_test
 
 import (
 	"reflect"
 	"testing"
 
-	"github.com/vmware-tanzu/carvel-kapp-controller/cmd/controller"
 	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	app_ "github.com/vmware-tanzu/carvel-kapp-controller/pkg/app"
```
Comment (danielhelfand) [inline pkg/app/apps_reconciler_test.go]: Can we use something besides `_` here? Maybe even just call appDep or appPkg?

File: cmd/controller/run.go [hunk 2]
```diff
@@ -70,157 +63,109 @@ func Run(opts Options, runLog logr.Logger) {
...
+		appFactory := app.AppFactory{coreClient, kcClient, kcConfig}
+		reconciler := app.NewAppsReconciler(kcClient, runLog.WithName("app"),
+			appFactory, refTracker, updateStatusTracker)
+
+		ctrl, err := controller.New("app", mgr, controller.Options{
 			Reconciler: NewUniqueReconciler(&ErrReconciler{
-				delegate: NewAppsReconciler(kcClient, runLog.WithName("ar"), appFactory, refTracker, updateStatusTracker),
-				log:      runLog.WithName("pr"),
+				delegate: reconciler,
+				log:      runLog.WithName("er"),
```
Comment (danielhelfand) [inline cmd/controller/run.go:116]: ar?

File: pkg/app/app_factory.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 VMware, Inc.
```
Comment (danielhelfand) [inline pkg/app/app_factory.go:1]: 2021?

**Comment by joe-kimmel-vmw** [comment]
those godocs linters got you good,  but the changes LGTM.     I do agree with Daniel's request for a more descriptive name than "app_" but i have no suggestions,  maybe because i'm missing what else called "app" we might collide with in that file.

File: cmd/controller/run.go (hunk 2 above)
Comment (cppforlife) [reply cmd/controller/run.go:116]: err reconciler

File: pkg/app/app_factory.go (hunk 3 above)
Comment (cppforlife) [reply pkg/app/app_factory.go:1]: i was under impression we keep existing copyright notices when files get moved around.

File: cmd/controller/run.go (hunk 2 above)
Comment (danielhelfand) [reply cmd/controller/run.go:116]: I see. The diff was throwing me off.

File: pkg/app/app_factory.go (hunk 3 above)
Comment (danielhelfand) [reply pkg/app/app_factory.go:1]: Yeah, not sure. Fine to keep unless we know otherwise.

File: pkg/app/apps_reconciler_test.go (hunk 1 above)
Comment (cppforlife) [reply pkg/app/apps_reconciler_test.go]: i think typically camelcase is not used in aliases. apppkg?

Comment (cppforlife) [reply pkg/app/apps_reconciler_test.go]: renamed to apppkg

**Pull request merged**